### ISSUE TITLE
Character Lengths of Paths

### ISSUE THREAD

**Issue by pgierz** [FESOM/fesom2#9]
Hi Fesom devs,

Several of the paths in `namelist.config` have predefined maximum lengths. I would recommend that they become arbitrarily long so the users can do what they want. e.g. this 
```fortran
character(5)           :: runid='test1'       ! a model/setup name
...
character(100)         :: MeshPath='./mesh/'
```

could be replaced with something that just reads the entire string.

That all happens in gen_modules_config.F90

**Comment by christian-stepanek** [comment]
I am not sure whether this already has been fixed, but in the current version of AWI-ESM2 (based on the FESOM2 version employed by Sidorenko et al., 2019) this is still a problem - and it is quite a pitfall as the error message does not really give an obvious hint towards a too long path and filename:

```
636: forrtl: No such file or directory
505: forrtl: No such file or directory
512: forrtl: No such file or directory
521: forrtl: No such file or directory
526: forrtl: No such file or directory
556: forrtl: severe (29): file not found, unit 99, file /work/ba1066/a270061/esm_experiments_v4//this_is_together_with_the_path_a_very_long_experiment_name_fesom.clock
556: Image              PC                Routine            Line        Source             
556: fesom              000000000076DADE  for__io_return        Unknown  Unknown
556: fesom              000000000078C015  for_open              Unknown  Unknown
556: fesom              000000000045BDF7  Unknown               Unknown  Unknown
556: fesom              000000000041134D  Unknown               Unknown  Unknown
556: fesom              000000000040D4BE  Unknown               Unknown  Unknown
556: libc-2.12.so       00002AE5BC544D20  __libc_start_main     Unknown  Unknown
556: fesom              000000000040D3A9  Unknown               Unknown  Unknown
534: forrtl: severe (29): file not found, unit 99, file /work/ba1066/a270061/esm_experiments_v4//this_is_together_with_the_path_a_v
```
I understand that it may not be of high priority, but in complex setups where many simulations are grouped into lengthy and complex paths this becomes a real usability issue.

**Comment by hegish** [comment]
I know, this is really annoying. I will see if I can fix it. Probably next week.

**Comment by hegish** [comment]
Do you know if this issue applies only to paths in namelist.config, or also to other paths (e.g. forcing files)?

**Comment by christian-stepanek** [comment]
Jan, I do not know for sure, but I think what happens is that some path variables are configured to be of fixed length, and this leads to problems for long paths and/or experiment names. I can prove this for the clock-file (see excerpt of the error log above). Maybe there are also other instances of input, that are messed up for long paths, and that would be coming up later on during initialization of the model.

As Paul noted, probably it would be sufficient to change the variables, that hold paths and names, to dynamical arrays - so that FESOM2 can handle strings of arbitrary length without cutting off parts of the path or simulation name. Here is an overview on where one could start with the fix, hopefully this already solves the issue for good:

lines 23-35 of gen_modules_config.F90:
```
 ! *** Paths for all in and out ***
  character(100)         :: MeshPath='./mesh/'
  character(100)         :: OpbndPath='./opbnd/'
  character(100)         :: ClimateDataPath='./hydrography/'
  character(100)         :: ForcingDataPath='./forcing/'
  character(100)         :: TideForcingPath='./tide_forcing/'
  character(100)         :: ResultPath='./result/'
  namelist /paths/  MeshPath, OpbndPath, ClimateDataPath, ForcingDataPath, &
       TideForcingPath, ResultPath

  ! *** ocean climatology data name ***
  character(100)         :: OceClimaDataName='annual_woa01_ts.out'
  logical                :: use_prepared_init_ice=.false.  !initialize ice externally
```

**Comment by hegish** [comment]
A quick but thorough fix is not possible here, as Fortran does not seem to be able to read character arrays of unknown length from namelists (i.e.). Thus I will have to redesign the whole namelist reading mechanism in order to provide a non-hackish solution. Anyway, I will try to address this soon.

**Comment by christian-stepanek** [comment]
If it is a major issue to make the arrays dynamic, would it be a quick and dirty "fix" to just allocate an array of a length that will likely never be met in practice, or that is close to a length limit defined by the operating system, e.g. 4096?
see here for example: https://stackoverflow.com/questions/9449241/where-is-path-max-defined-in-linux

This is of course just another workaround. But before one invests lots of time to overhaul the whole namelist reading process, maybe it is a practical solution to reduce the risk of headaches when setting up simulations.

**Comment by koldunovn** [comment]
It it's does not cause major drawbacks I would go for @christian-stepanek solution of just increasing the size to something sufficiently large. Of course the meaning of "sufficient" might change in the future, but let's be practical for now :)

**Comment by dguibert** [comment]
Hi,

Any advances on this issue?
Because I hit this issue and I have changed most of the path variables to a MAX_PATH length (equal to 1024).

I could review any PR or submit mine if you want to avoid anyone to do the modifications as well? What do you prefer?

**Comment by koldunovn** [comment]
@dguibert This was only fixed for result path by @patrickscholz (0a5f0d9b). If you could do the PR would be great! I finally added you to the repo, so you can create branch here if you prefer to.
